src/Symfony/Component/HttpFoundation/StreamedJsonResponse.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+/**
+ * StreamedJsonResponse represents a streamed HTTP response for JSON.
+ *
+ * A StreamedJsonResponse uses a structure and generics to create an
+ * efficient resource saving JSON response.
+ *
+ * It uses flush after a specified flush size to directly stream the data.
+ *
+ * @see flush()
+ *
+ * @author Alexander Schranz <alexander@sulu.io>
+ *
+ * Example usage:
+ *
+ * function loadArticles(): \Generator
+ *     // some streamed loading
+ *     yield ['title' => 'Article 1'];
+ *     yield ['title' => 'Article 2'];
+ *     yield ['title' => 'Article 3'];
+ * }),
+ *
+ * $response = new StreamedJsonResponse(
+ *     // json structure with generators in which will be streamed
+ *     [
+ *         '_embedded' => [
+ *             'articles' => loadArticles(), // any generator which you want to stream as list of data
+ *         ],
+ *     ],
+ * );
+ */
+class StreamedJsonResponse extends StreamedResponse
+{
+    public const DEFAULT_ENCODING_OPTIONS = JsonResponse::DEFAULT_ENCODING_OPTIONS;
+
+    private int $encodingOptions = self::DEFAULT_ENCODING_OPTIONS;
+
+    /**
+     * @param mixed[]                        $data      JSON Data containing PHP generators which will be streamed as list of data
+     * @param int                            $status    The HTTP status code (200 "OK" by default)
+     * @param array<string, string|string[]> $headers   An array of HTTP headers
+     * @param int                            $flushSize After every which item of a generator the flush function should be called
+     */
+    public function __construct(
+        private readonly array $data,
+        int $status = 200,
+        array $headers = [],
+        private int $flushSize = 100,
+    ) {
+        parent::__construct($this->stream(...), $status, $headers);
+
+        if (!$this->headers->get('Content-Type')) {
+            $this->headers->set('Content-Type', 'application/json');
+        }
+    }
+
+    private function stream(): void
+    {
+        $generators = [];
+        $structure = $this->data;
+
+        array_walk_recursive($structure, function (&$item, $key) use (&$generators) {
+            // generators should be used but for better DX all kind of Traversable are supported
+            if ($item instanceof \Traversable) {
```

Comment:
small idea I got. Maybe this should be `$item instanceof \Traversable && !$item instanceof \JsonSerializable` to avoid overriding the json representation of objects that are both traversable and json serializable.

Reply:
What about `ArrayObject` it is not `JsonSerializable` but `\json_encode(new \ArrayObject(["key" => "value"]));` would still converted to JSON 🤔

Reply:
perhaps \Traversable is slippery slope, since json_encode has some form of support for it ...

maybe it's better to only support \Generator and \IteretorAggregate, which mostly serve the lazy purpose and those actually do not work with json_encode: https://3v4l.org/DJghG

Reply:
`ArrayObject` seems also be a `\IteratorAggregate` 🤔 I now added support for `objects` based on if the first key of the array is `(int) 0` (not the best but only way how we can detect something like this on Generators). This way we solve the problem with the different array objects. See `testResponseOtherTraversable` Test cases.